pkg/rulefmt/rulefmt.go

```diff
@@ -193,10 +213,14 @@ func testTemplateParsing(rl *Rule) (errs []error) {
 	return errs
 }
 
+var grpsTmp ruleGroups
+
 // Parse parses and validates a set of rules.
 func Parse(content []byte) (*RuleGroups, []error) {
 	var groups RuleGroups
-	if err := yaml.UnmarshalStrict(content, &groups); err != nil {
+	err := yaml.Unmarshal(content, &groups)
+	_err := yaml.Unmarshal(content, &grpsTmp)
```

Comment:
Why are you doing this? This global also won't be threadsafe.

Reply:
we need another struct var containing similar names but with `yaml.Node`. `yaml.Node` is the only way we can print the line and column number. Hence we need a different var as well to get the line numbers for errors like `similar groupnames` and `empty groupname`.

Reply:
This isn't safe, nor is it good practice. Pass it via the objects you're returning.

Reply:
Done